Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -647,546 +552,351 @@ protected virtual void Dispose(bool disposing)
-                else
-                {
-                    // Internal interfaces must be private, not excluded and part of the LocalNetworkSubnet.
-                    _internalInterfaces = CreateCollection(_interfaceAddresses.Where(IsInLocalNetwork));
-                }
 
-                _logger.LogInformation("Defined LAN addresses : {0}", _lanSubnets.AsString());
-                _logger.LogInformation("Defined LAN exclusions : {0}", _excludedSubnets.AsString());
-                _logger.LogInformation("Using LAN addresses: {0}", _lanSubnets.Exclude(_excludedSubnets, true).AsNetworks().AsString());
+                result = NetworkExtensions.FormatIpString(availableInterfaces.First().Address);
+                _logger.LogDebug("{Source}: Using first internal interface as bind address: {Result}", source, result);
+                return result;
             }
+
+            // There isn't any others, so we'll use the loopback.
+            result = IsIpv4Enabled && !IsIpv6Enabled ? "127.0.0.1" : "::1";
+            _logger.LogWarning("{Source}: Only loopback {Result} returned, using that as bind address.", source, result);
+            return result;
         }
 
-        /// <summary>
-        /// Generate a list of all the interface ip addresses and submasks where that are in the active/unknown state.
-        /// Generate a list of all active mac addresses that aren't loopback addresses.
-        /// </summary>
-        private void InitialiseInterfaces()
+        /// <inheritdoc/>
+        public IReadOnlyList<IPData> GetInternalBindAddresses()
         {
-            lock (_intLock)
-            {
-                _logger.LogDebug("Refreshing interfaces.");
+            // Select all local bind addresses
+            return _interfaces.Where(x => IsInLocalNetwork(x.Address))
+                .OrderBy(x => x.Index)
+                .ToList();
+        }
 
-                _interfaceNames.Clear();
-                _interfaceAddresses.Clear();
-                _macAddresses.Clear();
+        /// <inheritdoc/>
+        public bool IsInLocalNetwork(string address)
+        {
+            if (NetworkExtensions.TryParseToSubnet(address, out var subnet))
+            {
+                return IPAddress.IsLoopback(subnet.Prefix) || (_lanSubnets.Any(x => x.Contains(subnet.Prefix)) && !_excludedSubnets.Any(x => x.Contains(subnet.Prefix)));
+            }
 
-                try
+            if (NetworkExtensions.TryParseHost(address, out var addresses, IsIpv4Enabled, IsIpv6Enabled))
+            {
+                bool match = false;
+                foreach (var ept in addresses)
                 {
-                    IEnumerable<NetworkInterface> nics = NetworkInterface.GetAllNetworkInterfaces()
-                        .Where(i => i.SupportsMulticast && i.OperationalStatus == OperationalStatus.Up);
+                    match |= IPAddress.IsLoopback(ept) || (_lanSubnets.Any(x => x.Contains(ept)) && !_excludedSubnets.Any(x => x.Contains(ept)));
```

Comment:
A Bitwise OR compound assignment is a bit weird looking and it doesn't short-circuit. Also, shouldn't you simply break out of the foreach if `match == true`?